backend/src/main/java/ca/bc/gov/app/service/client/ClientService.java

```diff
@@ -141,6 +142,79 @@ public Mono<ClientDetailsDto> getClientDetails(
             "Unable to process request. This sole proprietor is not owned by a person"
         )));
   }
+  
+  public Mono<ForestClientDetailsDto> getClientDetailsByClientNumber(
+      String clientNumber,
+      List<String> groups
+  ) {
+      log.info("Loading details for {} for user role {}", clientNumber, groups.toString());
+
+      return legacyService
+          .searchByClientNumber(clientNumber, groups)
+          .flatMap(forestClientDetailsDto -> {
+              String corpRegnNmbr = forestClientDetailsDto.corpRegnNmbr();
+
+              if (corpRegnNmbr == null || corpRegnNmbr.isEmpty()) {
+                  log.info("Corporation registration number not provided. Returning legacy details.");
+                  return Mono.just(forestClientDetailsDto);
+              }
+
+              log.info("Retrieved corporation registration number: {}", corpRegnNmbr);
+
+              return bcRegistryService
+                      .requestDocumentData(corpRegnNmbr)
+                      .next()
+                      .flatMap(documentMono -> 
+                            populateGoodStandingInd(forestClientDetailsDto, 
+                                                    documentMono)
+                      );
+          });
+  }
+
+  private Mono<ForestClientDetailsDto> populateGoodStandingInd(
+      ForestClientDetailsDto forestClientDetailsDto,
+      BcRegistryDocumentDto document
+  ) {
+      Boolean goodStandingInd = document.business().goodStanding();
+      String goodStanding;
+      if (goodStandingInd == null) {
+        goodStanding = "";
+      } 
+      else {
+        goodStanding = goodStandingInd ? "Y" : "N";
+      }
+
+      log.info("Setting goodStandingInd for client: {} to {}", 
+               forestClientDetailsDto.clientNumber(), goodStanding);
+
+      ForestClientDetailsDto updatedDetails = new ForestClientDetailsDto(
```

Comment:
If the `ForestClientDetailsDto` record is annotated with `@With` you can just call the `forestClientDetailsDto.withGoodStanding(goodStanding)` making the code simpler.

Reply:
Done! Thanks for the review